server/data/healthcheck.js

```diff
@@ -1,41 +1,57 @@
 /* eslint-disable prefer-promise-reject-errors */
 const superagent = require('superagent')
+const Agent = require('agentkeepalive')
+const { HttpsAgent } = require('agentkeepalive')
 const db = require('./dataAccess/db')
 const logger = require('../../log.js')
+const config = require('../config')
 
 function dbCheck() {
   return db.queryWithoutTransaction('SELECT 1 AS ok')
 }
 
-function serviceCheck(name, url) {
-  return new Promise((resolve, reject) => {
-    superagent
-      .get(url)
-      .timeout({
-        response: 4000,
-        deadline: 4500,
-      })
-      .end((error, result) => {
-        try {
-          if (error) {
-            logger.error(error.stack, `Error calling ${name}`)
-            return reject(`${error.status} | ${error.code} | ${error.errno}`)
-          }
+const agentOptions = {
+  maxSockets: config.apis.elite2.agent.maxSockets,
+  maxFreeSockets: config.apis.elite2.agent.maxFreeSockets,
+  freeSocketTimeout: config.apis.elite2.agent.freeSocketTimeout,
+}
 
-          if (result.status === 200) {
-            return resolve('OK')
-          }
+function serviceCheckFactory(name, url) {
+  const keepaliveAgent = url.startsWith('https') ? new HttpsAgent(agentOptions) : new Agent(agentOptions)
+  logger.warn(`We have constructed a new service check here for ${url}`)
```

Comment:
meh.. was this just for debug during development? do we need to log this

Reply:
Removed